src/Razor/src/Microsoft.VisualStudio.LanguageServerClient.Razor/ActivityLogRazorLogger.cs

```diff
@@ -49,6 +49,19 @@ public override void LogVerbose(string message)
             Log(__ACTIVITYLOG_ENTRYTYPE.ALE_INFORMATION, message);
         }
 
+        public async override void SetUIContext(Guid uiContextGuid, bool isActive)
```

Comment:
I'm not 100% sold on including this on `RazorLogger` since this isn't strictly "Logging", but it's messaging about the program state and this class already has good representation so we wouldn't have a ton of properties created just for the hypothetical `UIContextManager`.

Reply:
Ya I think "logging" probably isn't the right place for this functionality.  Also, just because we've published project context doesn't mean the Razor language server is "Ready" unfortunately.

Consider the following: 
- VS publishes a project.razor.json file.
- UI context set
- The server sees the project.razor.json file get created
    - Server starts understanding the project.razor.json

Because of this I'd suggest [making a custom target on the client](https://github.com/dotnet/aspnetcore-tooling/blob/master/src/Razor/src/Microsoft.VisualStudio.LanguageServerClient.Razor/DefaultRazorLanguageServerCustomMessageTarget.cs) and expand our language server to notify the client when it's "ready" (aka, it's absorbed a project.razor.json). Detecting project.razor.json absorption would be as easy as making a project snapshot change trigger that inspects the ProjectWorkspaceState and does a `null` check. It'd also isolate the "notification" to a single class on the server. When the client gets told the server is "ready" it could then activate the UI context.